index.js

```diff
@@ -104,80 +156,148 @@ exports.init = function (sbot, config) {
     if (rpc.id === sbot.id) return // ssb-client connecting to ssb-server
     if (isClient) {
       initialized.promise.then(() => {
-        const opts = { version: 3 }
-        const local = toPull.duplex(ebt.createStream(rpc.id, opts.version, true))
-        const remote = rpc.ebt.replicate(opts, (networkError) => {
-          if (networkError && getSeverity(networkError) >= 3) {
-            console.error('rpc.ebt.replicate exception:', networkError)
-          }
+        ebts.forEach((ebt) => {
+          const format = ebt.name
+          const opts = { version: 3, format }
+          const local = toPull.duplex(
+            ebt.createStream(rpc.id, opts.version, true)
+          )
+
+          // for backwards compatibility we always replicate classic
+          // feeds using existing replicate RPC
+          const replicate =
+            format === 'classic' ? rpc.ebt.replicate : rpc.ebt.replicateFormat
+
+          const remote = replicate(opts, (networkError) => {
+            if (networkError && getSeverity(networkError) >= 3) {
+              console.error('rpc.ebt.replicate exception:', networkError)
+            }
+          })
+          pull(local, remote, local)
         })
-        pull(local, remote, local)
       })
     }
   })
 
-  function request (destFeedId, requesting) {
+  function findEBTForFeed(feedId, formatName) {
+    let ebt
+    if (formatName) {
+      ebt = ebts.find((ebt) => ebt.name === formatName)
+    } else {
+      ebt = ebts.find((ebt) => ebt.isFeed(feedId))
+    }
+
+    if (!ebt) {
+      ebt = ebts.find((ebt) => ebt.name === 'classic')
+    }
+
+    return ebt
+  }
+
+  function request(destFeedId, requesting, formatName) {
     initialized.promise.then(() => {
-      if (!isFeed(destFeedId)) return
+      const ebt = findEBTForFeed(destFeedId, formatName)
+
+      if (!ebt.isFeed(destFeedId)) return
```

Comment:
@arj03 In the case of `formatName` being passed as `'indexed'`, then this line still suffers from that race condition because `isFeed` is going to look up the state from ssb-meta-feeds. I commented it out and thought things would work but still no luck. Might be caused by some other bug somewhere

Reply:
> Might be caused by some other bug somewhere

Ah, I think I know what it is. indexed format `isFeed` is called also in cleanClock, and in epidemic-broadcast-tree `request`, so we can't go around the race condition, gotta solve it.